integrated_channels/integrated_channel/transmitters/learner_data.py

```diff
@@ -381,9 +381,30 @@ def _log_exception_supplemental_data(self, learner_data, operation_name,
                 payload=learner_data
             )), exc_info=True)
 
-    def handle_transmission_error(self, learner_data, client_exception,
-                                  integrated_channel_name, enterprise_customer_uuid, learner_id, course_id):
-        """Handle the case where the transmission fails."""
+    def handle_transmission_error(self, learner_data, client_exception):
+        """
+        Subclasses who wish to do additional processing of transmission error
+        before logging, can do so in overrides of this method
+        """
+
+    def process_transmission_error(
```

Comment:
I am now using what is called the 'template' design pattern here meaning subclasses are only allowed to change parts of this method. In this case, logging always happens for free and subclasses don't need to call super().log_... anymore. If they do want additional processing of the error though, before logging, they can override the handle_ method (which only SAPSF does right now)